hack/check_version.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+#
+# Copyright (C) 2025 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# Usage: check_version.sh
+#
+# Implement 2 checks:
+# 1. check that the version numbers in galaxy.yml and rpm spec file are consistent
+#
+# 2. check major and minor version management according to our policy:
+# - If there is a removed role, check that the major version is incremented.
+# - If there is a new role, check that the minor version is incremented.
+
+error() {
+    if [ -n "${GITHUB_STEP_SUMMARY}" ]; then
+        echo "- check_version: ${1}" | tee -a "${GITHUB_STEP_SUMMARY}"
+    else
+        echo "Error: check_version: ${1}"
+    fi
+    exit 1
+}
+
+# Verify consistency of version numbers in galaxy.yml and rpm spec file
+spec_version=$(grep Version: ansible-collection-redhatci-ocp.spec | awk '{print $2}' | cut -d. -f1,2)
+galaxy_version=$(grep version: galaxy.yml | awk '{print $2}' | cut -d. -f1,2)
+
+if [ "$spec_version" != "$galaxy_version" ]; then
+    error "Inconsistent: rpm spec: ${spec_version} galaxy: ${galaxy_version}"
+fi
+
+branch=$(git rev-parse --abbrev-ref HEAD)
+
+if [ -z "$branch" ]; then
+    error "Unable to determine the current branch."
+fi
+
+# Get potentially removed roles
+REMOVED=$(git diff --diff-filter=D --name-only origin/main...$branch | grep -E '^roles/[^/]+/' | cut -d/ -f2 | uniq)
+ADDED=$(git diff --diff-filter=A --name-only origin/main...$branch | grep -E '^roles/[^/]+/' | cut -d/ -f2 | uniq)
```

Comment:
The grep is removing directories that could be part of a sub-role, as is the case for roles/acm/utils or roles/vendors/*

Reply:
Good catch. I have improved the code to take this into account. You can see the result when removing acm.utils here: https://github.com/redhatci/ansible-collection-redhatci-ocp/actions/runs/15131158226/job/42532362878?pr=678